Use ISO 8601 format for dates

KPJoshi opened the pull request:
In the USA, dates are often written as **month/day/year**. In many other countries, they are written as **day/month/year**. I believe that the second format is used for PAL region consoles. I propose using the ISO 8601 standard format for dates. This prints dates as **year-month-day**. For US region consoles, this will still be easy to understand since the year is the only 4 digit number and the order of month and day is preserved. For PAL region users, I believe my proposed format will be more understandable than the current default US format.

There might be 2 better alternatives:
- Representing the month as a 3 letter abbreviation. However in DS/3DS you enter the month as a number, so this will be slightly inconvenient, if less unambiguous.
- Adding a setting to decide the output format. But this would be out of the way in the dropdown menu.

Links:
- [ISO 8601 on Wikipedia](https://en.wikipedia.org/wiki/ISO_8601)
- [Date format by country on Wikipedia](https://en.wikipedia.org/wiki/Date_format_by_country)

Admiral-Fish commented:
This isn't the only place dates are used. A number of forms have date controls, shouldn't those be changed as well?

KPJoshi commented:
> This isn't the only place dates are used. A number of forms have date controls, shouldn't those be changed as well?

This is the function within the date/time file that is responsible for converting it to a string. So any place that displays the date uses this function. I checked in multiple locations in the compiled binary. Also I could not find any place where you enter the date yourself (except the year in some places)

Admiral-Fish commented:
There are several places in the UI that have selection for date and time. The constructors for DateEdit and DateTimeEdit should probably have a call to setDisplayFormat to follow ISO 8601.

KPJoshi commented:
> There are several places in the UI that have selection for date and time. The constructors for DateEdit and DateTimeEdit should probably have a call to setDisplayFormat to follow ISO 8601.

All the places where you can enter date or date+time _already_ use my proposed year-month-day format! Check, for example, the TID/SID finder for any gen. It seems that this is the default for QT when it comes to date and time inputs. I do not think adding any calls to setDisplayFormat will be required.

Admiral-Fish commented:
I think it defaults to the system locale. It shows as M/D/Y for me.

KPJoshi commented:
I added the formats to the constructors of DateEdit and DateTimeEdit. I also removed several inconsistencies:
- The Gen 4 TID/SID finder did not use the custom DateTimeEdit class but QT's original class. I changed it to use the custom class (otherwise the format would not be set)
- Some places had specified a conflicting date/time format that overrode the one provided in the constructor. I removed those conflicts.

Sorry for the multiple commits fixing the same problem. Please squash when you merge.

Admiral-Fish commented:
Will merge sometime this weekend. Been busy this week.
